lib/util/ts_parser.js

```diff
@@ -337,6 +337,68 @@ shaka.util.TsParser = class {
     return pes;
   }
 
+  /**
+   * Parse AVC Nalus
+   *
+   * @param {shaka.util.TsParser.PES} pes
+   * @return {!Array.<shaka.util.TsParser.AvcNalu>}
+   */
+  parseAvcNALus(pes) {
+    const timescale = shaka.util.TsParser.Timescale_;
+    const time = pes.pts ? pes.pts / timescale : null;
+    const data = pes.data;
+    const len = data.byteLength;
+    let state = 0;
+    const units = [];
+    let i = 0;
+    let lastUnitStart = -1;
+    let lastUnitType = 0;
+
+    while (i < len) {
+      const value = data[i++];
+      // optimization. state 0 and 1 are the predominant case. let's handle
+      // them outside of the switch/case
+      if (state == 0) {
+        state = value ? 0 : 1;
+        continue;
+      }
+      if (state == 1) {
+        state = value ? 0 : 2;
```

Comment:
These states are unreadable as ints.  Can you name them and give them enums?

Is this based on code from somewhere else?  If so, even if we're changing it in review, please add comments to give proper credit.

Reply:
Added the credit